src/flake8_requirements/checker.py

```diff
@@ -288,18 +301,41 @@ def parse_options(cls, options):
                 for x in re.split(r"],?", options.known_modules)[:-1]
             ]
         }
+        cls.requirements_max_depth = options.requirements_max_depth
 
     @classmethod
     @memoize
     def get_requirements(cls):
         """Get package requirements."""
-        try:
-            with open("requirements.txt") as f:
-                return tuple(parse_requirements(f.readlines()))
-        except IOError as e:
-            LOG.debug("Couldn't open requirements file: %s", e)
+        if not os.path.exists("requirements.txt"):
+            LOG.debug("No requirements.txt file")
             return ()
 
+        return tuple(parse_requirements(cls.resolve_requirement(
+            "-r requirements.txt", cls.requirements_max_depth + 1)))
+
+    @classmethod
+    def resolve_requirement(cls, requirement, max_depth=0):
+        """Resolves flags like -r in an individual requirement line."""
+        if requirement.startswith("#") or not requirement.strip():
+            return []
+
+        if requirement.startswith("-r "):
```

Comment:
Does pip work when there is a space at the beginning? If yes, you should `strip()` before `startswith()` and then perform only `requirement[3:].lstrip()`.